src/prefect/_vendor/README.md

```diff
@@ -0,0 +1,26 @@
+# Vendored dependencies of `prefect`
+
+## FastAPI, vendored from tiangolo/fastapi@0.103.2
+
+We are temporarily vendoring `fastapi` to help us untangle `pydantic` requirements as we
+gradually bring `pydantic>2` support to `prefect`.  FastAPI's compatibility layer
+makes a strong assumption that having `pydantic>2` installed means that all of your
+models are v2 models.  We will be adjusting that compatibility layer accordingly.
+
+If we do need to update the version before we're done with it:
+
+```
+VERSION=0.103.2
+rm -Rf /tmp/fastapi
+git clone https://github.com/tiangolo/fastapi -b $VERSION /tmp/fastapi
+rsync -ravdP /tmp/fastapi/fastapi/ src/prefect/_vendor/fastapi/
```

Comment:
Thank you for this! Can 100% see a future where we need to update our internal fastapi version before we fully deprecate pydantic v1.